src/store/tag-type/actions.js

```diff
@@ -0,0 +1,73 @@
+import api from './api';
+
+export const START_FETCH_TAG_TYPES = 'START_FETCH_TAG_TYPES';
+export const RECEIVE_TAG_TYPES = 'RECEIVE_TAG_TYPES';
+export const ERROR_FETCH_TAG_TYPES = 'ERROR_FETCH_TAG_TYPES';
+export const START_CREATE_TAG_TYPE = 'START_CREATE_TAG_TYPE';
+export const ADD_TAG_TYPE = 'ADD_TAG_TYPE';
+export const ERROR_CREATE_TAG_TYPE = 'ERROR_CREATE_TAG_TYPE';
+export const START_DELETE_TAG_TYPE = 'START_DELETE_TAG_TYPE';
+export const DELETE_TAG_TYPE = 'DELETE_TAG_TYPE';
+export const ERROR_DELETE_TAG_TYPE = 'ERROR_DELETE_TAG_TYPE';
+export const START_UPDATE_TAG_TYPE = 'START_UPDATE_TAG_TYPE';
+export const UPDATE_TAG_TYPE = 'UPDATE_TAG_TYPE';
+export const ERROR_UPDATE_TAG_TYPE = 'ERROR_UPDATE_TAG_TYPE';
+
+function receiveTagTypes(json) {
+    return {
+        type: RECEIVE_TAG_TYPES,
+        value: json,
+        receivedAt: Date.now(),
+    };
+}
+
+function dispatchAndThrow(dispatch, type) {
+    return error => {
+        dispatch({ type, error, receivedAt: Date.now() });
+        // throw error;
```

Comment:
The throw needs to actually throw to make sure the submit function in view component is able to catch errors from storing.